### PR TITLE
632 dynamic tab investigation

### LINKED ISSUE #632: Dynamic Tab Investigation
https://leetcode.com/problems/maximum-nesting-depth-of-the-parentheses/?envType=daily-question&envId=2024-04-04

Leetcode got a really cool feature, where it alow you to arrange every component's position, adjust, merge tab. We want to support that in our code .

### PR DESCRIPTION
**Pull request by Dammshine**

### CODE REVIEW

**Review by NE1NN — APPROVED** [review]
lgtm

File: client/package.json
Comment (dqna64) [inline client/package.json:1]: are these packages being used

File: client/src/components/TabResize/DynamicTabs.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+    const handleMouseMove = (moveEvent: MouseEvent) => {
+      const innerScopeIndex = index;
+
+      const currentPos = isHorizontal ? moveEvent.clientX : moveEvent.clientY;
+      const delta =
+        (currentPos - startPos) / (isHorizontal ? window.innerWidth : window.innerHeight);
+      const ratioAfterMouseDown = [...ratioBeforeMouseDown];
+
+      // TODO: THIS IS IMPORTANT
```
Comment (dqna64) [inline client/src/components/TabResize/DynamicTabs.tsx:50]: what  needs to be done here

**Review by dqna64 — COMMENTED** [review]
very nice

**Pull request closed without merging**